
syntax = "proto3";

package user;

service UserService {
    rpc Register (RegisterRequest) returns (RegisterResponse) {}
    rpc Login (LoginRequest) returns (LoginResponse) {}
}

message RegisterRequest {
    string username = 1;
    string email = 2;
    string password = 3;
}

message RegisterResponse {
    string message = 1;
    bool success = 2;
}

message LoginRequest {
    string email = 1;
    string password = 2;
}

message LoginResponse {
    string message = 1;
    bool success = 2;
    string token = 3;
    UserInfo user = 4;
}

message UserInfo {
    string id = 1;
    string username = 2;
    string email = 3;
}
